src/map/skill.cpp

```diff
@@ -20855,6 +20855,30 @@ static bool skill_check_unit_movepos(uint8 check_flag, struct block_list *bl, sh
 	return unit_movepos(bl, dst_x, dst_y, easy, checkpath);
 }
 
+/**
+ * Get skill duration after adjustments by skill_duration mapflag
+ * @param src: Source map location
+ * @param skill_id: Skill ID
+ * @param skill_lv: Skill level
+ * @return Adjusted skill duration
+ */
+int skill_get_time3(int16 m, uint16 skill_id, uint16 skill_lv) {
+	int time = 0;
+
+	if (!skill_get_index(skill_id) || !(time = skill_get_time(skill_id, skill_lv)) || m < 0)
+		return 0;
+	
+	struct map_data *mapdata = map_getmapdata(m);
+
+	if (!mapdata->skill_duration.empty()) {
+		for (const auto &it : mapdata->skill_duration) {
+			if (it.first == skill_id)
+				return time * it.second / 100;
```

Comment:
* change the `time * it.second / 100;` to `time / 100 * it.second;` worried about overflow after the multiplication and beside the time is in msec witch div'd by 100 always should be good.
* change to use `find` instead the iterator? since it's note vector anymore, (looks cleaner xD)